### PR TITLE
LPS-143276 Fix invalid url warning

### PR DESCRIPTION
**Pull request by thektan**
https://issues.liferay.com/browse/LPS-143276

Mainly looking for a sanity check. Thanks!

### CODE REVIEW

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/utils/fetch.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+describe('fetch', () => {
+	describe('addParams', () => {
+		it('adds a parameter to a url', () => {
+			expect(
+				addParams('/o/search-experiences-rest/v1.0/sxp-elements', {
+					test: 'value',
+				}).href
+			).toEqual(
+				'http://localhost:8080/o/search-experiences-rest/v1.0/sxp-elements?test=value'
```
Comment (oliv-yu) [inline modules/dxp/apps/search-experiences/search-experiences-web/test/js/utils/fetch.js]: Just curious, how do the tests know that Liferay.ThemeDisplay.getPortalURL is localhost:8080? Is that the same when running tests on servers?

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/mocks/fetch.js [hunk 2]
```diff
@@ -11,10 +11,22 @@
 
 import {INDEX_FIELDS, QUERY_SXP_ELEMENTS, mockClassNames} from './data';
 
+function trimURL(url) {
+	let trimmedURL = url.href ? url.href : url; // Handling if url is a URL object or a string
```
Comment (oliv-yu) [inline modules/dxp/apps/search-experiences/search-experiences-web/test/js/mocks/fetch.js]: Oh does `url.href || url` work too?

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/edit_sxp_blueprint/EditSXPBlueprintForm.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ afterAll(() => {
 	console.error = originalError;
 });
 
-Liferay.ThemeDisplay = {getDefaultLanguageId: () => 'en_US'};
+Liferay.ThemeDisplay.getDefaultLanguageId = () => 'en_US';
```
Comment (oliv-yu) [inline modules/dxp/apps/search-experiences/search-experiences-web/test/js/edit_sxp_blueprint/EditSXPBlueprintForm.js]: There's a line like this in the PageToolbar.js test, I can add that in too if that's ok?

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/utils/fetch.js (hunk 1 above)
Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/test/js/utils/fetch.js]: This is the `jest.config.js` that `liferay-npm-scripts test` uses: https://github.com/liferay/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/config/jest.config.js

And if you follow the `setupFiles: [require.resolve('../jest/setup.js')],`, this is where the mock is defined: https://github.com/liferay/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/jest/mocks/Liferay.js#L126

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/mocks/fetch.js (hunk 2 above)
Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/test/js/mocks/fetch.js]: Thanks I'll make this change.

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/edit_sxp_blueprint/EditSXPBlueprintForm.js (hunk 3 above)
Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/test/js/edit_sxp_blueprint/EditSXPBlueprintForm.js]: I'll make this change too.

**Comment by thektan** [comment]
Thanks for the review! I'll make the changes you noted and send it to BChan.

**Comment by thektan** [comment]
Squashed the changes into the original commits. Forwarded https://github.com/brianchandotcom/liferay-portal/pull/110642

**Pull request closed without merging**